packages/sirius-web/frontend/sirius-web-application/src/views/edit-project/workbench-views/details/DetailsView.tsx

```diff
@@ -92,28 +97,32 @@ export const DetailsView = forwardRef<WorkbenchViewHandle, WorkbenchViewComponen
 
     const skip = state.objectIds.length === 0;
     const { payload, complete, loading } = useDetailsViewSubscription(editingContextId, state.objectIds, skip);
-
     useEffect(() => {
       if (isFormRefreshedEventPayload(payload)) {
         setState((prevState) => ({ ...prevState, form: payload.form }));
+        fetchLabels(editingContextId, state.objectIds);
       }
     }, [payload]);
 
-    const { classes } = useDetailsViewStyles();
+    const { classes } = useViewStyles();
+
     const detailsViewConfiguration = initialConfiguration as unknown as DetailsViewConfiguration;
     const initialSelectedPageId = detailsViewConfiguration?.selectedPageId ?? null;
 
     const toolbar = (
-      <Tooltip
-        title={
-          state.pinned
-            ? 'Pinned. Click to follow the global selection.'
-            : 'Unpinned. Click to pin the view to the current selection.'
-        }
-        data-testid="details-toggle-pin"
-        onClick={() => setState((prevState) => ({ ...prevState, pinned: !prevState.pinned }))}>
-        {state.pinned ? <SyncLockOutlinedIcon /> : <SyncOutlinedIcon />}
-      </Tooltip>
+      <>
+        {state.pinned && !skip ? <Typography variant="caption">{'Pinned on: ' + labels}</Typography> : null}
+        <Tooltip
+          title={
+            state.pinned
+              ? 'Pinned. Click to follow the global selection.'
+              : 'Unpinned. Click to pin the view to the current selection.'
+          }
+          data-testid="details-toggle-pin"
+          onClick={() => setState((prevState) => ({ ...prevState, pinned: !prevState.pinned }))}>
+          {state.pinned ? <SyncLockOutlinedIcon /> : <SyncOutlinedIcon />}
+        </Tooltip>
+      </>
```

Comment:
If those tens lines are repeated three or four times, it can be a simple component like a `SynchronizationButton` or something like that, and it should properly be the one fetching the labels.